zstacklib/zstacklib/hardware/pci/address.py

```
@@ -0,0 +1,81 @@
+        PCI address string like "0000:00:1f.0".
+
+    Raises:
+        PciError: When pci_device is missing required fields.
+    """
+    try:
+        domain = _to_int(pci_device.get("domain", 0))
+        bus = _to_int(pci_device["bus"])
+        slot = _to_int(pci_device["slot"])
+        function = _to_int(pci_device["function"])
+    except KeyError as exc:
+        raise PciError("missing pci address field: %s" % exc)
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)